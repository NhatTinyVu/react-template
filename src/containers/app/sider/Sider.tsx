import React from 'react';
import { Link, RouteComponentProps } from 'react-router-dom';
import useRouter from 'use-react-router';
import { get } from 'lodash';
import { Layout, Menu, Icon } from 'antd';
import { routes, Routes } from '../constants';

const initialKey = get(routes, '[0].url', '');

interface ItemProps {
  title: string;
  url: string;
}

export const Item: React.FC<ItemProps> = ({ title, url }: ItemProps) => (
  <span>
    <Icon type="unordered-list" />
    <span>
      <Link to={url}>{title}</Link>
    </span>
  </span>
);

const Sider: React.FC = () => {
  const {
    location: { pathname },
  }: RouteComponentProps = useRouter();

  const selectedKey = get(
    Routes.find(({ url }) => url === pathname),
    'url',
    initialKey,
  );

  return (
    <Layout.Sider breakpoint="md" collapsedWidth="0">
      <Menu
        mode="inline"
        theme="dark"
        defaultSelectedKeys={[initialKey]}
        selectedKeys={[selectedKey]}
      >
        {routes.map(({ title, url, childrens }) => (
          <Menu.SubMenu key={url} title={<Item title={title} url={url} />}>
            {childrens &&
              childrens.map(route => (
                <Menu.Item key={route.url}>
                  <Link to={route.url}>{route.title}</Link>
                </Menu.Item>
              ))}
          </Menu.SubMenu>
        ))}
      </Menu>
    </Layout.Sider>
  );
};

export default Sider;
