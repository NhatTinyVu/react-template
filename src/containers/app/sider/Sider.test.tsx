import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Sider from '.';

it('renders correctly', (): void => {
  const wrapper = shallow(<Sider />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
