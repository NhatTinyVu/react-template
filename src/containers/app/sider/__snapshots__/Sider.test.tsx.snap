// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Sider
  breakpoint="md"
  collapsedWidth="0"
>
  <Menu
    defaultSelectedKeys={
      Array [
        "/react-hooks",
      ]
    }
    mode="inline"
    selectedKeys={
      Array [
        "/react-hooks",
      ]
    }
    theme="dark"
  >
    <SubMenu
      key="/react-hooks"
      title={
        <Item
          title="React hooks"
          url="/react-hooks"
        />
      }
    >
      <MenuItem
        key="/react-hooks/use-state"
      >
        <Link
          to="/react-hooks/use-state"
        >
          useState
        </Link>
      </MenuItem>
    </SubMenu>
  </Menu>
</Sider>
`;

exports[`renders correctly 2`] = `
<span>
  <Icon
    type="unordered-list"
  />
  <span>
    <Link
      to="url"
    >
      title
    </Link>
  </span>
</span>
`;
